sdk/samples/iot/iot_samples_common.c

```diff
@@ -1,7 +1,28 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // SPDX-License-Identifier: MIT
 
-#include "iot_sample_foundation.h"
+#ifdef _MSC_VER

```

Comment:
this file should include all the includes from the header. I think ones like:
```c
#include <stdbool.h>
#include <stddef.h>
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>
```
are missing.

Reply:
So I think this is where the question is that @ahsonkhan had, and I see his point: should the c file duplicate what the h file already has, given the c and h file have the same name?  I think this becomes redundant.